rules/okta/credential_access_okta_brute_force_or_password_spraying.toml

```diff
@@ -0,0 +1,49 @@
+[metadata]
+creation_date = "2020/07/16"
+ecs_version = ["1.5.0"]
+maturity = "production"
+updated_date = "2020/07/16"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies a high number of failed Okta user authentication attempts from a single IP address, which could be indicative
+of a brute force or password spraying attack. An adversary may attempt a brute force or password spraying attack to
+obtain unauthorized access to user accounts.
+"""
+false_positives = ["Automated processes that attempt to authenticate using expired credentials and unbounded retries may lead to false positives."]
+index = ["filebeat-*"]
+language = "kuery"
+license = "Elastic License"
+name = "Okta Brute Force or Password Spraying Attack"
+references = [
+    "https://developer.okta.com/docs/reference/api/system-log/",
+    "https://developer.okta.com/docs/reference/api/event-types/",
+]
+risk_score = 47
+rule_id = "42bf698b-4738-445b-8231-c834ddefd8a0"
+severity = "medium"
+tags = ["Elastic", "Okta"]
+type = "threshold"
+
+query = '''
+event.module:okta and event.dataset:okta.system and event.category:authentication and event.outcome:failure
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1110"
+name = "Brute Force"
+reference = "https://attack.mitre.org/techniques/T1110/"
+
+
+[rule.threat.tactic]
+id = "TA0006"
+name = "Credential Access"
+reference = "https://attack.mitre.org/tactics/TA0006/"
+
+[rule.threshold]
+field = "source.ip"
+value = 25
```

Comment:
Just a comment, this seems like a lot of failed attempts. If this is a good measuring stick based on your experience, no need to change it - I've not investigated how many times a brute force/spray happens before success.

Reply:
Thanks @peasead. The default lockout policy for Okta users is 10 failed authentication attempts. If an automated or brute force attack is happening, I expect the number of failed authentication attempts to be quite high.

I'm planning on developing another couple of rules to detect attempts to gain unauthorized access to Okta accounts. E.g. https://github.com/elastic/detection-rules/issues/73 and https://github.com/elastic/detection-rules/issues/74. We can always revisit this threshold value.